src/app/host/[room]/page.tsx

```diff
@@ -1,46 +1,150 @@
-import HostControls from '@/components/HostControls'
 import RoomMainUi from '@/components/RoomMainUi'
+import RoomInfo from '@/components/RoomInfo'
 
-export default function HostRooms({ params }: { params: { room: string } }) {
-	const { room } = params
+export default function HostRoom({ params }: { params: { room: string } }) {
+	const [allUserPointData, setAllUserPointData] = useState<ListenerRes[]>([])
+	const storyPointRes = useSocketListener('story-points')
+	const joinRoomRes = useSocketListener('join-room')
 
-	console.log('%c>>> HOST ROOM roomId:', 'color: #5f0', room)
+	useEffect(() => {
+		if (!storyPointRes) return
+
+		console.log('%c>>> listener', 'color: red', storyPointRes)
+		setAllUserPointData((prevUsersPoints) => {
+			const index = prevUsersPoints.findIndex((data) => {
+				return data.userName === storyPointRes.userName
+			})
+			if (index !== -1) {
+				const noDuplicates = [...prevUsersPoints]
+				noDuplicates[index].message = storyPointRes.message
+				console.log('%c>>> noDuplicates SP', 'color: red', noDuplicates)
+				return noDuplicates
+			}
+			return [...prevUsersPoints, storyPointRes]
+		})
+	}, [storyPointRes])
+
+	useEffect(() => {
+		if (!joinRoomRes) return
+
+		console.log('%c>>> joinRoomRes', 'color: red', joinRoomRes)
+
+		setAllUserPointData((prevUsersPoints) => {
+			const index = prevUsersPoints.findIndex((data) => {
+				return data.userName === joinRoomRes.userName
+			})
+			if (index !== -1) {
+				const noDuplicates = [...prevUsersPoints]
+				noDuplicates[index].message = joinRoomRes.message
+				console.log('%c>>> noDuplicates JR', 'color: red', noDuplicates)
+				return noDuplicates
+			}
+			return [...prevUsersPoints, joinRoomRes]
+		})
+	}, [joinRoomRes])
+
+	const { room } = params
 
 	const hostDataLocalStorage = localStorage.getItem('scrumDivingHostData')
-	const hostData = hostDataLocalStorage && JSON.parse(hostDataLocalStorage)
+	const hostName =
+		hostDataLocalStorage && JSON.parse(hostDataLocalStorage)?.hostName
 
-	console.log(
-		'%c>>> HOST ROOM hostData from LocalStorage',
-		'color: yellow',
-		hostData,
-	)
-	const hostFirstName = hostData?.hostName
+	useEffect(() => {
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter('join-room', 'join', hostName, timeStamp, room)
+	}, [room, hostName])
 
 	useEffect(() => {
-		socketRoomEmitter('join-room', 'Room Joined', room, hostFirstName)
-	}, [room, hostFirstName])
+		console.log(
+			'%c>>> allUserPointData Host Room',
+			'color: #f0f',
+			allUserPointData,
+		)
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter(
+			'all-users-story-points',
+			allUserPointData,
+			hostName,
+			timeStamp,
+			room,
+		)
+		localStorage.setItem(
+			'scrumDivingStoryPoints',
+			JSON.stringify([allUserPointData]),
+		)
+	}, [allUserPointData, room, hostName])
+
+	const handleShowPoints = () => {
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter(
+			'show-disable-reset-points',
+			'true',
```

Comment:
question: Why is this `"true"` as string instead of boolean? Are you fighting TypeScript? I feel like you are... 👀

Reply:
Because the listener returns a string. Yes, I'm sure I could modify it to return a boolean, but we use this pattern a lot at work where everything coming from the server is an array, an array of objects, and all the values are strings so as not to have any worries about data type. All our booleans are "1" or "0". I went with "true" or "false" just to be more explicit, but same basic idea. And, yes, I don't have to fight with TypeScript to narrow the type from string | boolean | ListenerRes[]. 

Honestly, Sandrina, fighting TypeScript feels like a huge waste of time. TS is very useful and helpful, but it is so picky and there is so much difficulty in dealing with it, that it isn't worth the time to me if I have something that works and is type safe. I'm more than happy to have a string value that can be turned into a boolean, and turning (variable === 'true') into a boolean is a no brainer. 

So I'm not so much fighting TS as using it in a way that works.

Reply:
> and all the values are strings to be consistent. Even prices are strings!

Oh wow... Does that sound right to you? That's an odd pattern. I mean, it's possible, but don't take as granted that it will be like that everywhere else. 

> I would have to narrow the type down, and doing so has been a huge pain.

Yes it is, don't tell me about it. I agree with you, it can feel like a huge waste of time, especially for pet-projects. All good to keep it as string and move on. Just make it explicit in your code "TODO: ideally should be a boolean, but TS is complex." and move on.

Reply:
I made the change to a boolean using your `as unknown as TYPE` override. I like that better.

And yes, the first time I saw that we were passing prices as strings, I was really surprised. I think it has something to do with our shitty eCommerce server, WebSphere Commerce Server. 

Now that I think about it, I probably should assume that many things I do at work are NOT what other companies will do. Like pass around strings everywhere. We do so many things that are inconsistent. And weird. And probably dumb. LOL